build/pause/Makefile

```diff
@@ -12,20 +12,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY:	build push
+IMAGE = gcr.io/google_containers/pause-$(ARCH)
+TAG = 3.0
 
-TAG = 2.0
+CFLAGS = -Os -Wall -static
+ARCH ?= amd64
+KUBE_CROSS_VERSION ?= $(shell cat ../build-image/cross/VERSION)
 
-ARCH = amd64
-IMAGE = pause
-# ARCH = arm
-# IMAGE = pause-arm
+BIN = pause
+SRCS = pause.c
 
-build:
-	./prepare.sh ${ARCH}
-	docker build -t gcr.io/google_containers/$(IMAGE):$(TAG) .
+ifeq ($(ARCH),amd64)
+	TRIPLE ?= x86_64-linux-gnu
+endif
 
-push:	build
-	gcloud docker --server=gcr.io push gcr.io/google_containers/$(IMAGE):$(TAG)
+ifeq ($(ARCH),arm)
+	TRIPLE ?= arm-linux-gnueabi
+endif
 
-all:	push
+all: .push
+
+$(BIN): $(SRCS)
+	docker run -v $(CURDIR):/build gcr.io/google_containers/kube-cross:$(KUBE_CROSS_VERSION) \
+		/bin/bash -c "\
+			cd /build && \
+			$(TRIPLE)-gcc $(CFLAGS) -o pause $^ && \
+			$(TRIPLE)-strip $@"
+
+container .container: pause
+	docker build -t $(IMAGE):$(TAG) .
```

Comment:
maybe it's better to use a `REGISTRY?=gcr.io/google_containers` variable instead
Also, the `amd64` image maybe should be tagged `gcr.io/google_containers/pause:3.0` for backwards compability. @thockin WDYT?

See example:
https://github.com/kubernetes/kubernetes/blob/master/cluster/images/hyperkube/Makefile